src/Compilers/Core/Portable/Text/SourceTextComparer.cs

```diff
@@ -1,15 +1,17 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
 using System.Collections.Generic;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Text
 {
-    internal class SourceTextComparer : IEqualityComparer<SourceText>
+    internal class SourceTextComparer : IEqualityComparer<SourceText?>
     {
         public static SourceTextComparer Instance = new SourceTextComparer();
```

Comment:
Interesting. This should have been `readonly`, but it's not picked up by the analyzer because it's public.